### PR TITLE
types: use discord-api-types `Locale`

### PR DESCRIPTION
**Pull request by Syjalo**
**Please describe the changes this PR makes and why it should be merged:**

This PR replaces locale types definitions from `string` to `Locale` from discord-api-types.

**Status and versioning classification:**

Please move lines that apply to you out of the comment:
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by SpaceEEC — APPROVED** [review]

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -1328,8 +1329,8 @@ export class Interaction<Cached extends CacheType = CacheType> extends Base {
   public user: User;
   public version: number;
   public memberPermissions: CacheTypeReducer<Cached, Readonly<PermissionsBitField>>;
-  public locale: string;
-  public guildLocale: CacheTypeReducer<Cached, string, string, string>;
+  public locale: Locale;
+  public guildLocale: CacheTypeReducer<Cached, Locale>;
```
Comment (Jiralite) [inline packages/discord.js/typings/index.d.ts:1333]: What's the reason for this always being possibly `null`?

**Review by Jiralite — CHANGES_REQUESTED** [review]

File: packages/discord.js/typings/index.d.ts [hunk 2]
```diff
@@ -60,6 +60,7 @@ import {
   GuildNSFWLevel,
   GuildPremiumTier,
   GuildVerificationLevel,
+  Locale,
```
Comment (vladfrangu) [inline packages/discord.js/typings/index.d.ts:63]: Can you export this in index.js too?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/discord.js/typings/index.d.ts (hunk 2 above)
Comment (Syjalo) [reply packages/discord.js/typings/index.d.ts:63]: Done

File: packages/discord.js/src/index.js [hunk 3]
```diff
@@ -184,6 +184,7 @@ exports.GuildVerificationLevel = require('discord-api-types/v9').GuildVerificati
 exports.InteractionType = require('discord-api-types/v9').InteractionType;
 exports.InteractionResponseType = require('discord-api-types/v9').InteractionResponseType;
 exports.InviteTargetType = require('discord-api-types/v9').InviteTargetType;
+exports.Locale = require('discord-api-types/v9').Locale;
```
Comment (ImRodry) [inline packages/discord.js/src/index.js:187]: this isn't re-exported

Comment (Syjalo) [reply packages/discord.js/src/index.js:187]: Thanks

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**